xbmc/video/dialogs/GUIDialogSubtitles.h

```diff
@@ -50,6 +50,12 @@ class CGUIDialogSubtitles : public CGUIDialog, CJobQueue
   void Download(const CFileItem &subtitle);
   void OnDownloadComplete(const CFileItemList *items, const std::string &language);
 
+  /*!
+   * Called when the context menu is requested on a subtitle service
+   * present on the list of installed subtitle addons
+   * */
+  void OnSubtitleServiceContextMenu(const int itemIdx);
```

Comment:
`const`makes no sense here.

Reply:
Why so if the value is supposed to never be changed?

Reply:
There is zero benefit in using `const` for by-value parameters, but there are downsides. If you later want to change the *implementation* that it shall change the value of the parameter, because you want to avoid introduction of another local var just to copy the const paramter value to be able to manipulate it, you have to change the header and to recompile everything dependent. Hassle without any need.

Reply:
Well while your comment makes great sense as a generic rule of thumb I don't see how it can have much relevance in this particular context. The likelihood of having another dialog extending the subtitlesdialog class is pretty much inexistent. Furthermore this refers to the index of the item (on a list) which the context menu has just been requested, it is not supposed to be changed by the implementation (nor I foresee any plausible reason for it in the future). It's much more likely that someone, for some reason, wants to pass a const listitem reference.
If you agree, I'd prefer to keep it as is for now.

Reply:
> If you agree, I'd prefer to keep it as is for now.

 Well, I do not really agree.

 Which value has the information for the reader of the header, that the copy (!) of the parameter's value they pass to the method will not be modified by the implementation of the method? We're not talking about references here, which would be a completely different story. We are talking about values that are copied onto the stack, then passed to the method. `const`is pointless here because the called method cannot manipulate the data the caller passes to the method.

Reply:
I see what you mean now, since this will result in a copy of the original value it's pointless being declared const as we shouldn't care what the implementation does with it. I'll just remove `const` tomorrow as I don't think it makes much sense to pass it as a const ref.   
Thanks for pointing it out

Reply:
you are welcome. :-)